lib/heapwatch.js

```diff
@@ -0,0 +1,53 @@
+"use strict";
+
+var cluster = require('cluster');
+
+function HeapWatch (conf, logger, statsd) {
+    this.conf = conf =  conf || {};
+    this.limit = (conf.limitMB || 1500) * 1024 * 1024;
+    this.logger = logger;
+    this.statsd = statsd;
+    this.checkInterval = 60000; // once per minute
+    this.failCount = 0;
+}
+
+HeapWatch.prototype.watch = function() {
+    var usage = process.memoryUsage();
+
+    // Report memory stats to statsd. Use 'timing' (which isn't really
+    // timing-specific at all) instead of 'gauge' to get percentiles &
+    // min/max.
+    var statsd = this.statsd;
+    statsd.timing('heap.rss', usage.rss);
+    statsd.timing('heap.total', usage.heapTotal);
+    statsd.timing('heap.used', usage.heapUsed);
+
+    if (usage.heapUsed > this.limit) {
+        console.log(usage.heapUsed, this.limit);
```

Comment:
Why not use `this.logger.log()` (or remove this entirely since something is going to be logged either way) ?